docs/sources/flow/reference/components/loki.process.md

```diff
@@ -221,6 +294,236 @@ stage {
+the value stored in the shared map under that name.
+
+Let's see what happens when the following log line is put through this
+two-stage pipeline:
+```
+{"timestamp":"2022-01-01T01:00:00.000000001Z"}
+
+stage {
+  json {
+    expressions = { time = "timestamp" }
+  }
+}
+stage {
+  regex {
+    expression = "^(?P<year>\\d+)"
+    source     = "time"
+  }
+}
+```
+
+The first stage would add the following key-value pairs into the extracted map:
+```
+time: 2022-01-01T01:00:00.000000001Z
+```
+
+Then, the regex stage would parse the value for time from the shared values and
+append the following key-value pairs back into the extracted values map:
+```
+year: 2022
+```
+
+
+### timestamp block
+
+The `timestamp` inner block configures a processing stage that sets the
+timestamp of log entries before they're forwarded to the next component. When
+no timestamp stage is set, the log entry timestamp defaults to the time when
+the log entry was scraped.
+
+The following arguments are supported:
+
+Name                | Type           | Description                                                 | Default   | Required
+------------------- | -------------- | ----------------------------------------------------------- | --------- | --------
+`source`            | `string`       | Name from extracted values map to use for the timestamp.    |           | yes
+`format`            | `string`       | Determines how to parse the source string.                  |           | yes
+`fallback_formats`  | `list(string)` | Fallback formats to try if the `format` field fails.        | `[]`      | no
+`location`          | `string`       | IANA Timezone Database location to use when parsing.        | `""`      | no
+`action_on_failure` | `string`       | What to do when the timestamp can't be extracted or parsed. | `"fudge"` | no
+
+The `source` field defines which value from the shared map of extracted values
+the stage should attempt to parse as a timestamp.
+
+The `format` field defines _how_ that source should be parsed.
+
+First off, the `format` can be set to one of the following shorthand values for
+commonly-used forms:
+```
+ANSIC: Mon Jan _2 15:04:05 2006
+UnixDate: Mon Jan _2 15:04:05 MST 2006
+RubyDate: Mon Jan 02 15:04:05 -0700 2006
+RFC822: 02 Jan 06 15:04 MST
+RFC822Z: 02 Jan 06 15:04 -0700
+RFC850: Monday, 02-Jan-06 15:04:05 MST
+RFC1123: Mon, 02 Jan 2006 15:04:05 MST
+RFC1123Z: Mon, 02 Jan 2006 15:04:05 -0700
+RFC3339: 2006-01-02T15:04:05-07:00
+RFC3339Nano: 2006-01-02T15:04:05.999999999-07:00
+
+Additionally, support for common Unix timestamps is supported with the following format values:
+Unix: 1562708916 or with fractions 1562708916.000000123
+UnixMs: 1562708916414
+UnixUs: 1562708916414123
+UnixNs: 1562708916000000123
+```
+
+Otherwise, the field accepts a custom format string that defines how an
+arbitrary reference point in history (Mon Jan 2 15:04:05 -0700 MST 2006) should
+be interpreted by the stage.
```

Comment:
```suggestion
Otherwise, the field accepts a custom format string that defines how an
arbitrary reference point in history (for example, `Mon Jan 2 15:04:05 -0700 MST 2006`) should
be interpreted by the stage.
```

Reply:
This is not an example, should I rephrase it somehow?

The Go language uses this set point time in history `Mon Jan 2 15:04:05 -0700 MST 2006`, as a reference point when parsing time formats.

So if I wanted to format my dates as "YYYY-MM-DD" I should use the reference time to denote that i.e. "2006-01-02". Similarly, if I wanted to format it as "HH:MM", I'd use "03:04 pm".

Reply:
Wow, weird. Should you say something like: 

```suggestion
Otherwise, the field accepts a custom format string that defines how an
arbitrary reference point in history should
be interpreted by the stage. The arbitrary reference point is Mon Jan 2 15:04:05 -0700 MST 2006.
```